### DISCUSSION TITLE
Enable/Disable autoevents

### DISCUSSION THREAD

**Discussion by vijay-hcl-epl** [edgexfoundry/community#6] [Q&A] [answered]
Hi all..
We are using edgex in one of our products. One of our requirements is to dynamically enable/disable the data reception from end devices. 

Currently we are using auto-events to receive data from the end devices. So to enable/disable data reception dynamically from the code we are looking for some APIs or curl commands in edgex. Please let me know if any such API/curl/any other feature is available in edgex to enable/disable auto-events dynamically from the code? 

We are seeing some APIs for device-sdk-c(edgex_device_autoevent_start). Similar kind of APIs are available for device-sdk-go? Please give your comments..Thank you in advance..!!

**Comment by lenny-goodell** [comment, marked as answer]
Please check out Core Metadata API for updating device data
[patch_device](https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-metadata/2.2.0#/default/patch_device)

**Reply by cloudxxx8** [reply]
You can remove the autoevent or just modify the adminState to LOCK.  They can both be set by the API Lenny provided (PATCH Device).

**Reply by vijay-hcl-epl** [reply]
Thank you Lenny and cloud for the suggestions. By locking the admin state I was able to stop the auto events. But I think it also makes the device unresponsive. Is that right? Also I tried removing autoevents using the patch_device API with the curl command, I was not able to remove the autoevents. I just removed the autoevents key value pair in the '-d' section of the curl command but it did not work out. Could you please let me know if I am missing anything? Thank you

**Reply by cloudxxx8** [reply]
yes, adminState Lock means the device service won't talk to the device.
How do you use the PATCH to remove the autoevents?  You need to pass an empty array to it.

**Reply by vijay-hcl-epl** [reply]
Thank you Cloud..after passing an empty array it worked..!! earlier I removed the entire autoevents array and the keyword "autoevents". Now everything is fine..Thank you again..!!

**Comment by vijay-hcl-epl** [comment]
Using patch_device API with below 2 settings worked..!! Thanks Lenny and Cloud..!!
1. adminstate can be used (OR)
2. pass an empty array for autoevents
